test/functional/wallet_disableprivatekeys.py

```diff
@@ -28,5 +29,15 @@ def run_test(self):
         assert_raises_rpc_error(-4,"Error: Private keys are disabled for this wallet", w1.getrawchangeaddress)
         w1.importpubkey(w2.getaddressinfo(w2.getnewaddress())['pubkey'])
 
+        self.log.info('Test that private keys cannot be imported')
+        addr = w2.getnewaddress('')
+        privkey = w2.dumpprivkey(addr)
+        assert_raises_rpc_error(-4, 'Cannot import private keys to a wallet with private keys disabled', w1.importprivkey, privkey)
+        result = w1.importmulti([{'scriptPubKey': {'address': addr}, 'timestamp': 'now', 'keys': [privkey]}])
+        assert(not result[0]['success'])
+        assert('warning' not in result[0])
+        assert_equal(result[0]['error']['code'], -4)
+        assert_equal(result[0]['error']['message'], 'Cannot import private keys to a wallet with private keys disabled')
+
```

Comment:
@UdjinM6 15235 do we need smth like https://github.com/bitcoin/bitcoin/pull/15235/files#diff-63c7710d72381cffe692175dd70e47f1891bf75c3e09249ed826f9df3917d02dR3833-R3836 ?

Reply:
I'm not sure what you mean... (the link doesn't point to any specific line for me)

Reply:
https://github.com/bitcoin/bitcoin/pull/15235/files# file src/wallet/rpcwallet.cpp line 3833-3836

Reply:
Ah, I see. Yeah, it might be a good idea to add smth like
```
    if (pwallet->IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS)) {
        throw JSONRPCError(RPC_WALLET_ERROR, "Error: Private keys are disabled for this wallet");
    }
```
to `upgradetohd` rpc in `rpcwallet.cpp`, and maybe also to `importelectrumwallet` rpc in `rpcdump.cpp`. Good catch! 👍 (could be done in a follow-up PR though)

Reply:
added at both places as mentioned, please review